src/harness/sas_test_harness.py

```diff
@@ -0,0 +1,311 @@
+      ppa_record_dump_file_path = os.path.join(self.getDumpFilePath(), generated_file_name)
+      # Check cbsdReferenceID in ppaInfo and add cbsdReferenceID if not present
+      if not ppa_record['ppaInfo'].has_key('cbsdReferenceId'):
+         ppa_record['ppaInfo'].update({'cbsdReferenceId': ['cbsdId-1', 'cbsdId-2']})
+      writeDumpFile(ppa_record_dump_file_path, ppa_record)
+      ppa_record_url = self.getSasThUrl() + '/' + ppa_record['id']
+      fad_record_ppa = createFadRecord(ppa_record_url, ppa_record)
+      fad_record_list.append(fad_record_ppa)
+
+    for esc_sensor_record in esc_sensor_records:
+      #generated_file_name = '{}.json'.format(esc_sensor_record['id'].split('/')[-1])
+      generated_file_name = '{}.json'.format(str(hashlib.sha1(esc_sensor_record['id']).hexdigest()).encode('utf-8'))
+      esc_sensor_record_dump_file_path = os.path.join(self.getDumpFilePath(), generated_file_name)
+      writeDumpFile(esc_sensor_record_dump_file_path, esc_sensor_record)
+      esc_sensor_record_url = self.getSasThUrl() + '/%s/%s' \
+                                                      % (esc_sensor_record['id'], generated_file_name)
+      fad_record_esc = createFadRecord(esc_sensor_record_url, esc_sensor_record)
+      fad_record_list.append(fad_record_esc)
+
+    fad_file_path = os.path.join(self.getDumpFilePath(), 'FAD.json')
+    full_activity_dump = createFadObject(fad_record_list)
+    writeDumpFile(fad_file_path, full_activity_dump)
+    logging.info("FAD Records generated successfully.The files are placed "
+                 "under the path:%s", self.getDumpFilePath())
+
+
+class SasTestHarnessServerHandler(BaseHTTPRequestHandler):
+  """SasTestHarnessServerHandler class is inherited with BaseHTTPRequestHandler
+  to serve HTTP Response
+  """
+  @classmethod
+  def setupDumpFilePath(cls, file_path):
+    """Setup the absolute path of dumpfile used in do_GET() method """
+    cls.__dumpfile_path = file_path
+    if not os.path.exists(cls.__dumpfile_path):
+      os.makedirs(cls.__dumpfile_path, 0755)
+    cls.__fadfile_path = os.path.join(cls.__dumpfile_path, 'FAD.json')
+
+  def do_GET(self):
+    """Handles Pull/GET Request and returns Path of the Request to callback Method"""
+
+    def checkDumpFileExist(url_id):
+      """vCheck the url_id received in GET request passed as argument to check corresponding
+      dump file exist in that path
+      """
+      return os.path.exists(os.path.join(self.__dumpfile_path, '{}.json'.format(url_id)))
+
+    def getDumpFilePath(url_id):
+      """ Return the absolute path of FAD records that are going
+      to be generated
+      """
+      return os.path.join(self.__dumpfile_path, '{}.json'.format(url_id))
+
+    if "dump" in self.path:
+      self.send_response(200)
+      self.send_header('Content-type', 'application/json')
+      self.end_headers()
+      self.wfile.write(json.load(open(os.path.join(self.__fadfile_path))))
```

Comment:
Currently the test cases create the files in a local directory and serves those files when requested. These are done in a linear fashion and not simultaneously at moment in the test cases. We made it this based on an earlier comment from that is copied below that mentions that files have to be written locally and then served from there.


kate-harrison on Jan 9 
Contributor
In addition to the full dump message itself, we will also need to send back the files containing the CBSDs, PPAs, and ESC sensors. At first glance, this method doesn't appear to support that.
Edit: I see that you're setting up a separate instance for each piece of data that's included in the SAS TH. I'm not sure this is a sustainable long-term solution and it's a bit hard to read. I thought the solution we discussed at the F2F was to give the SAS TH a local directory from which it would serve files (if there's a local file whose name matches the requested path, it's served; otherwise HTTP 404). Then we can simply swap out files locally and leave the SAS TH running for the duration of the test. Is there a reason you chose to deviate from that?